### PR TITLE
chore: update API endpoints from api.tez.ie to ecadinfra.com

### PR DESCRIPTION
**Pull request by GImbrailo**
This commit updates all instances of the old API endpoint (api.tez.ie) to the new one (ecadinfra.com). This change is necessary to align with the updated infrastructure and ensure all API calls are directed to the correct server.

chore: update RPC URLs from api.tez.ie to ecadinfra.com

This commit updates all instances of the old RPC URLs (api.tez.ie) to the new ones (ecadinfra.com) across multiple files. This change is necessary to keep the project up-to-date with the current infrastructure.

chore(docs): update TezosToolkit API endpoint URLs across multiple files

The API endpoint URLs for TezosToolkit instances have been updated from 'https://*.api.tez.ie' to 'https://*.ecadinfra.com' in multiple files. This change is made to reflect the updated API endpoint URLs provided by ECAD Labs.

chore(docs): update TezosToolkit URLs to use ecadinfra.com domain

The URLs for TezosToolkit instances in the documentation have been updated from api.tez.ie and ghostnet.api.tez.ie to mainnet.ecadinfra.com and ghostnet.ecadinfra.com respectively. This change is made to reflect the new domain for the Tezos infrastructure provided by ECAD Labs.

### CODE REVIEW

**Comment by AlirezaHaghshenas** [comment]
Is this a good chance to move this to a single file, to reduce repetition? @dsawali @GImbrailo

File: docs/rpc_nodes.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ values={[
 
 | Provider         | Net          | URL                                      | Header                                                                          |
 |------------------|--------------|------------------------------------------|---------------------------------------------------------------------------------|
-| ECAD Labs        | Mainnet      | https://mainnet.api.tez.ie               | [Check](https://mainnet.api.tez.ie/chains/main/blocks/head/header)              |
+| ECAD Labs        | Mainnet      | https://mainnet.ecadinfra.com               | [Check](https://mainnet.ecadinfra.com/chains/main/blocks/head/header)              |
```
Comment (hui-an-yang) [inline docs/rpc_nodes.md:26]: mainnet header link isn't working https://mainnet.ecadinfra.com/chains/main/blocks/head/header

Comment (GImbrailo) [reply docs/rpc_nodes.md:26]: That was an infrastructure issue that has since been resolved. Thanks for picking that up @hui-an-yang

**Review by hui-an-yang — APPROVED** [review]
lgtm 🎉

**Review by AlirezaHaghshenas — APPROVED** [review]

**Review by dsawali — APPROVED** [review]

**Pull request merged**